foundationdb/src/options.rs

```diff
@@ -0,0 +1,615 @@
+    ///
+    /// Transforms ``key`` using a versionstamp for the transaction. Sets the transformed key in the database to ``param``. A versionstamp is a 10 byte, unique, monotonically (but not sequentially) increasing value for each committed transaction. The first 8 bytes are the committed version of the database. The last 2 bytes are monotonic in the serialization order for transactions. WARNING: At this time versionstamps are compatible with the Tuple layer only in the Java and Python bindings. Note that this implies versionstamped keys may not be used with the Subspace and Directory layers except in those languages.
+    SetVersionstampedKey(Vec<u8>),
+    /// value to versionstamp and set
+    ///
+    /// Transforms ``param`` using a versionstamp for the transaction. Sets ``key`` in the database to the transformed parameter. A versionstamp is a 10 byte, unique, monotonically (but not sequentially) increasing value for each committed transaction. The first 8 bytes are the committed version of the database. The last 2 bytes are monotonic in the serialization order for transactions. WARNING: At this time versionstamped values are not compatible with the Tuple layer.
+    SetVersionstampedValue(Vec<u8>),
+    /// value to check against database value
+    ///
+    /// Performs lexicographic comparison of byte strings. If the existing value in the database is not present, then ``param`` is stored. Otherwise the smaller of the two values is then stored in the database.
+    ByteMin(Vec<u8>),
+    /// value to check against database value
+    ///
+    /// Performs lexicographic comparison of byte strings. If the existing value in the database is not present, then ``param`` is stored. Otherwise the larger of the two values is then stored in the database.
+    ByteMax(Vec<u8>),
+}
+
+impl MutationType {
+    pub fn code(&self) -> fdb::FDBMutationType {
+        match *self {
+            MutationType::Add(ref _v) => 2,
+            MutationType::And(ref _v) => 6,
+            MutationType::BitAnd(ref _v) => 6,
+            MutationType::Or(ref _v) => 7,
+            MutationType::BitOr(ref _v) => 7,
+            MutationType::Xor(ref _v) => 8,
+            MutationType::BitXor(ref _v) => 8,
+            MutationType::Max(ref _v) => 12,
+            MutationType::Min(ref _v) => 13,
+            MutationType::SetVersionstampedKey(ref _v) => 14,
+            MutationType::SetVersionstampedValue(ref _v) => 15,
+            MutationType::ByteMin(ref _v) => 16,
+            MutationType::ByteMax(ref _v) => 17,
+        }
+    }
+}
+
+pub enum ConflictRangeType {
+    /// Used to add a read conflict range
+    Read,
+    /// Used to add a write conflict range
+    Write,
+}
+
+impl ConflictRangeType {
+    pub fn code(&self) -> fdb::FDBConflictRangeType {
+        match *self {
+            ConflictRangeType::Read => 0,
+            ConflictRangeType::Write => 1,
+        }
+    }
+}
+
+pub enum ErrorPredicate {
```

Comment:
Can you look at `error::FdbError` in the `foundationdb` crate? We should maybe remove my hand-rolled version.

Reply:
I replaced `FdbErrorPredicate` with `options::ErrorPredicate`. I'm not sure about name of `fn code() ->  fdb::FDBErrorPredicate`. Its okay for other enum types which might have seperated value like `String`, bug for `ErrorPredicate` case `fn value() -> ...` sounds better.